src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -667,7 +667,11 @@ def _get_index(self):
             The union of all indexes across the partitions.
         """
         # TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset
-        return self._query_compiler.index
+        from snowflake.snowpark.modin.plugin.extensions.index import Index
+
+        if self._query_compiler.is_multiindex():
+            return self._query_compiler.index
+        return Index(data=self, convert_to_lazy=True)
```

Comment:
```suggestion
        return Index(self)
```
No need to emphasize `convert_to_lazy=True` this should be default behavior.